### PR TITLE
[LuaRocks] package version badge

### PR DESCRIPTION
**Pull request by un-def**
This is an reopening of #766.
I've updated and rebased my changes onto upstream and added tests.

`/luarocks/v/USERNAME/PACKAGE.FORMAT`

### CODE REVIEW

File: lib/luarocks-version.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Utilities relating to Lua 'rocks' version numbers.
+ * This compares version numbers using the algorithm
+ * followed by luarocks command-line utility
+ */
+'use strict';
+
+// Compare two arrays containing splitted and transformed to
+// positive/negative numbers parts of version strings,
+// respecting negative/missing values:
+// [1, 2, 1] > [1, 2], but [1, 2, -1] < [1, 2] ([1, 2] is aligned to [1, 2, 0])
```
Comment (paulmelnikow) [inline lib/luarocks-version.js]: Would it be worth adding a unit test with these examples?

File: lib/text-formatters.js [hunk 2]
```diff
@@ -45,3 +45,12 @@ function metric(n) {
...
 }
 exports.metric = metric;
+
+// Remove the starting v in a string.
+function omitv(version) {
+  if (version.charCodeAt(0) === 118) {
+    return version.slice(1);
+  }
+  return version;
+}
```
Comment (paulmelnikow) [inline lib/text-formatters.js]: There is a duplicate implementation of this function in `lib/php-version.js`. This seems like a better place for it to live. Could you remove that one?

File: service-tests/luarocks.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  }));
+
+t.create('unknown package')
+  .get('/v/nil/does-not-exist.json')
+  .expectJSON({ name: 'luarocks', value: 'invalid' });
+
+t.create('connection error')
+  .get('/v/mpeterv/luacheck.json')
+  .networkOff()
+  .expectJSON({ name: 'luarocks', value: 'inaccessible' });
```
Comment (paulmelnikow) [inline service-tests/luarocks.js]: These are lovely.

File: lib/text-formatters.js (hunk 2 above)
Comment (un-def) [reply lib/text-formatters.js]: Okay, I've replaced `lib/php-version.js` `omitv` function declaration with import from `lib/text-formatters.js`

File: lib/luarocks-version.js (hunk 1 above)
Comment (un-def) [reply lib/luarocks-version.js]: You're absolutely right! I've added unit tests… and found a bug! :)

File: lib/luarocks-version.spec.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      {versionString: '2.0-beta', expected: [2, 0, -3000]},
+      {versionString: '2.0-beta5', expected: [2, 0, -2995]},
+    ];
+    tests.forEach(function(test) {
+      const {versionString, expected} = test;
+      it(`parse '${versionString}' to [${expected}]`, function() {
+        assert.deepEqual(parseVersion(versionString), expected);
+      });
+    });
+  });
```
Comment (paulmelnikow) [inline lib/luarocks-version.spec.js]: Very thorough! These are nice.

**Pull request merged**